dspace-server-webapp/src/test/java/org/dspace/app/rest/ResourcePolicyRestRepositoryIT.java

```diff
@@ -2723,4 +2727,299 @@ public void resourcepolicySerchMethodWithPluralModelTest() throws Exception {
+                             .andExpect(status().isUnprocessableEntity());
+    }
+
+    @Test
+    public void updateResourcePolicyOfEPersonWithEmptyTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicy rpOfEPerson = ResourcePolicyBuilder.createResourcePolicy(context)
+                                                          .withDspaceObject(community)
+                                                          .withAction(Constants.READ)
+                                                          .withUser(eperson)
+                                                          .build();
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(put("/api/authz/resourcepolicies/" + rpOfEPerson.getID() + "/eperson")
+                             .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                             .content(StringUtils.EMPTY))
+                             .andExpect(status().isUnprocessableEntity());
+    }
+
+    @Test
+    public void updateResourcePolicyOfEPersonWithMultipleEPersonsTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson eperson1 = EPersonBuilder.createEPerson(context)
+                                         .withEmail("eperson1@mail.com")
+                                         .withPassword(password)
+                                         .build();
+        EPerson eperson2 = EPersonBuilder.createEPerson(context)
+                                         .withEmail("eperson2@mail.com")
+                                         .withPassword(password)
+                                         .build();
+
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicy rpOfEPerson = ResourcePolicyBuilder.createResourcePolicy(context)
+                                                          .withDspaceObject(community)
+                                                          .withAction(Constants.READ)
+                                                          .withUser(eperson)
+                                                          .build();
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(put("/api/authz/resourcepolicies/" + rpOfEPerson.getID() + "/eperson")
+                             .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                             .content("/api/eperson/epersons/" + eperson1.getID() +
+                                      "\n/api/eperson/epersons/" + eperson2.getID()))
+                             .andExpect(status().isUnprocessableEntity());
+    }
```

Comment:
I might be missing it, but I think we are missing tests to verify that non-Admin users **cannot** successfully change a ResourcePolicy via these endpoints.  All the existing tests look to be using Admin authentication at all times.